packages/block-editor/src/components/inserter/search-items.js

```diff
@@ -44,94 +43,98 @@ const removeMatchingTerms = ( unmatchedTerms, unprocessedTerms ) => {
+			return (
+				intersectionWith(
+					normalizedSearchTerms,
+					normalizeSearchTerm( variation.title ),
+					( termToMatch, labelTerm ) =>
+						labelTerm.includes( termToMatch )
+				).length > 0
+			);
+		} );
+		// When no patterns matched, fallback to all variations.
+		if ( isEmpty( matchedVariations ) ) {
+			return item;
+		}
+
+		return {
+			...item,
+			variations: matchedVariations,
+		};
+	} );
+};
+
 /**
  * Filters an item list given a search term.
  *
  * @param {Array} items       Item list
- * @param {Array} categories  Available categories.
- * @param {Array} collections Available collections.
  * @param {string} searchTerm Search term.
- *
- * @return {Array}             Filtered item list.
+ * @param {Object} config     Search Config.
+ * @return {Array}            Filtered item list.
  */
-export const searchItems = ( items, categories, collections, searchTerm ) => {
+export const searchItems = ( items, searchTerm, config = {} ) => {
 	const normalizedSearchTerms = normalizeSearchTerm( searchTerm );
-
 	if ( normalizedSearchTerms.length === 0 ) {
 		return items;
 	}
 
-	return items
-		.filter(
-			( { name, title, category, keywords = [], variations = [] } ) => {
-				let unmatchedTerms = removeMatchingTerms(
-					normalizedSearchTerms,
-					title
-				);
-
-				if ( unmatchedTerms.length === 0 ) {
-					return true;
-				}
-
-				unmatchedTerms = removeMatchingTerms(
-					unmatchedTerms,
-					keywords.join( ' ' )
-				);
-
-				if ( unmatchedTerms.length === 0 ) {
-					return true;
-				}
-
-				unmatchedTerms = removeMatchingTerms(
-					unmatchedTerms,
-					get( find( categories, { slug: category } ), [ 'title' ] )
-				);
-
-				const itemCollection = collections[ name.split( '/' )[ 0 ] ];
-				if ( itemCollection ) {
-					unmatchedTerms = removeMatchingTerms(
-						unmatchedTerms,
-						itemCollection.title
-					);
-				}
-
-				if ( unmatchedTerms.length === 0 ) {
-					return true;
-				}
-
-				unmatchedTerms = removeMatchingTerms(
-					unmatchedTerms,
-					variations
-						.map( ( variation ) => variation.title )
-						.join( ' ' )
-				);
-
-				return unmatchedTerms.length === 0;
-			}
-		)
-		.map( ( item ) => {
-			if ( isEmpty( item.variations ) ) {
-				return item;
-			}
-
-			const matchedVariations = item.variations.filter( ( variation ) => {
-				return (
-					intersectionWith(
-						normalizedSearchTerms,
-						normalizeSearchTerm( variation.title ),
-						( termToMatch, labelTerm ) =>
-							labelTerm.includes( termToMatch )
-					).length > 0
-				);
-			} );
-			// When no partterns matched, fallback to all variations.
-			if ( isEmpty( matchedVariations ) ) {
-				return item;
-			}
-
-			return {
-				...item,
-				variations: matchedVariations,
-			};
-		} );
+	const defaultGetTitle = ( item ) => item.title;
+	const defaultGetKeywords = ( item ) => item.keywords || [];
+	const defaultGetCategory = ( item ) => item.category;
+	const defaultGetCollection = () => null;
```

Comment:
Why not `() => []` for collection to align with keywords and variations? :)

Reply:
Because it's "one collection" so it aligns  more with category or title